examples/platform/esp32/mode-support/static-supported-modes-manager.cpp

```diff
@@ -31,13 +31,15 @@ using SemanticTag          = Structs::SemanticTagStruct::Type;
 template <typename T>
 using List = app::DataModel::List<T>;
 
-const StaticSupportedModesManager StaticSupportedModesManager::instance = StaticSupportedModesManager();
+SupportedModesManager::ModeOptionsProvider * StaticSupportedModesManager::epModeOptionsProviderList = nullptr;
 
-SupportedModesManager::ModeOptionsProvider StaticSupportedModesManager::epModeOptionsProviderList[FIXED_ENDPOINT_COUNT];
+const StaticSupportedModesManager StaticSupportedModesManager::instance = StaticSupportedModesManager();
 
-void StaticSupportedModesManager::InitEndpointArray()
+void StaticSupportedModesManager::InitEndpointArray(int size)
 {
-    for (int i = 0; i < FIXED_ENDPOINT_COUNT; i++)
+    mSize                     = size;
+    epModeOptionsProviderList = new SupportedModesManager::ModeOptionsProvider[mSize];
```

Comment:
Should this be deleted somewhere?  

Is InitEndpointArray() guaranteed to be called only once?

Reply:
> Should this be deleted somewhere?

We are deleting it in the destructor.

> Is InitEndpointArray() guaranteed to be called only once?

Yes, InitEndpointArray() should be called only once in the application.

Reply:
> We are deleting it in the destructor.

Where?

Reply:
> Yes, InitEndpointArray() should be called only once in the application.

That is very much not documented.

Reply:
> > We are deleting it in the destructor.
> 
> Where?

If you check `~StaticSupportedModesManager()`, we are calling `FreeSupportedModes` for every allocated endpoint

Reply:
> > Yes, InitEndpointArray() should be called only once in the application.
> 
> That is very much not documented.

Ok. I'll add comment for this.

Reply:
> we are calling FreeSupportedModes

Where does that delete the array that InitEndpointArray allocates?

Reply:
Memory allocated in InitEndpointArray will be needed for lifetime of the program.

Reply:
Then we should document, the reasons we are not freeing it anywhere.

Reply:
Added comment [here](https://github.com/project-chip/connectedhomeip/pull/27341/files#diff-a4009156ee3afaf0eab155be1527536b3c7af1992ffbad3bbd113b8d3d0143a3R43)